examples/vision/image_ocr.py

```diff
@@ -211,6 +212,12 @@ def __init__(self, monogram_file, bigram_file, minibatch_size,
         self.blank_label = self.get_output_size() - 1
         self.absolute_max_string_len = absolute_max_string_len
 
+        # Previously included in self.on_train_begin
```

Comment:
Why the switch?

Reply:
Another method, `next_train()`, relies on the results of these steps. But, `self.on_train_begin()` is only called at the beginning of `model.fit()` ([see explanation here.](https://www.tensorflow.org/guide/keras/custom_callback#an_overview_of_callback_methods)) So, [this line](https://github.com/MLH-Fellowship/keras-io/blob/f2d7aa8b8201eefc56cf6218b3ed16c0b847e125/examples/vision/image_ocr.py#L549) fails, because `next_train()` is called before `model.fit()` is called.

I've clarified the wording in the comment in another commit.

Reply:
So I would do two things:

1) pull that other commit with the updated comment in to this PR (since it's relevant);
2) leave `on_train_begin()` intact but just make sure it's called once before any call to next_train is made.

~

One way to accomplish 2):

in __init__, put some kind of sentinel variable, e.g.

```
  [...]
  self._wordlist_was_built = False
  [...]
```

Then, in `on_train_begin`, set that variable to `True`. Then, in `next_train`, you can check the state of `_wordlist_was_built` and, if it's `False`, call `on_train_begin`.

~

The motivation here is to ensure that nothing unexpected happens. With your implementation (`on_train_begin` runs in `__init__`), if somebody were to use `model.fit` instead of `model.fit_generator`, `on_train_begin` would run twice (once in `__init__` and once when `model.fit` explicitly calls it). That could cause unexpected consequences.

Reply:
For 1, is there something else I need to to do pull the commits into this PR? Right now, I'm seeing them already underneath this discussion. ("joshuacwnewton added 3 commits...") But, I may be misunderstanding how PRs work.

For 2, I like what you've proposed. It preserves much more of the original implementation than my changes did. Thank you for making such a detailed suggestion. :)

Reply:
Aha, I can link commits by their hash! That should make referencing them easier.

I've implemented your suggestions for 2 in commit 9e97d65. :)